src/nwm_routing/src/nwm_routing/__main__.py

```diff
@@ -1177,6 +1195,22 @@ def main_v03(argv):
 
             if data_assimilation_parameters:
                 lastobs_df = new_lastobs(run_results, dt * nts)
+                lastobs_df.index.names = ['link']
+                lastobs_df.insert(0, 'last_model_discharge', q0.loc[lastobs_df.index]['qu0'])
+                lastobs_df.insert(0, 'gages', gages)
+                lastobs_output_folder = data_assimilation_parameters.get('lastobs_output_folder',False)
+                if lastobs_output_folder:
+                    lastobs_string = lastobs_output_folder+"lastobs_df_"+str(run_set_iterator)+".nc"
+                else:
+                    lastobs_string = "lastobs_df_"+str(run_set_iterator)+".nc"
+                if not 'modelTimeAtOutput' in lastobs_df.columns:
+                    lastobs_df.insert(loc=0, column='modelTimeAtOutput', value=(time.time() - main_start_time))
+                if not 'Nudge' in lastobs_df.columns:
+                    lastobs_df.insert(loc=0, column='Nudge', value=np.NaN)
+                lastobs_df['Nudge'] = np.NaN
+                lastobs_df['modelTimeAtOutput'] = (time.time() - main_start_time)
+                lastobs_df.to_xarray().to_netcdf(lastobs_string)
+                # lastobs_df = pd.read_csv(lastobs_string,index_col='link')
```

Comment:
It seems like we are doing the same thing twice with the code block up top and this one here. Could they be consolidated?